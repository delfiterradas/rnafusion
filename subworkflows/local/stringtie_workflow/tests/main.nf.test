nextflow_workflow {

    name "Test STRINGTIE_WORKFLOW"
    script "../main.nf"
    workflow "STRINGTIE_WORKFLOW"
    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "stringtie"

    test("Should run stringtie workflow with stringtie = true") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                        [ id:'test', strandedness:'reverse' ], // meta map
                    file("${params.test_data_base}/genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam", checkIfExists: true)
                ])
                input[1] = Channel.of([
                    [ id:'test' ], // meta map
                    file("${params.test_data_base}/genomics/homo_sapiens/genome/genome.gtf", checkIfExists: true)
                ])
                """
            }
            params {
                stringtie = true
                outdir = "$outputDir"
                test_data_base = "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() },
            )
        }
    }

    test("Should run stringtie workflow with params.all = true") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                        [ id:'test', strandedness:'reverse' ], // meta map
                    file("${params.test_data_base}/genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam", checkIfExists: true)
                ])
                input[1] = Channel.of([
                    [ id:'test' ], // meta map
                    file("${params.test_data_base}/genomics/homo_sapiens/genome/genome.gtf", checkIfExists: true)
                ])
                """
            }
            params {
                all = true
                outdir = "$outputDir"
                test_data_base = "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() },
            )
        }
    }

}
