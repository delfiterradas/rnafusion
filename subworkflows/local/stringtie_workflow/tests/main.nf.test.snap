{
    "Should run stringtie workflow with params.all = true": {
        "content": [
            {
                "0": [
                    "stringtie.merged.gtf:md5,6da479298d73d5b3216d4e1576a2bdf4"
                ],
                "1": [
                    "versions.yml:md5,16bf76d2f56399280213cfa721195317",
                    "versions.yml:md5,16bf76d2f56399280213cfa721195317",
                    "versions.yml:md5,be2acf8efb04dc33562c9d00df7c3a50"
                ],
                "stringtie_gtf": [
                    "stringtie.merged.gtf:md5,6da479298d73d5b3216d4e1576a2bdf4"
                ],
                "versions": [
                    "versions.yml:md5,16bf76d2f56399280213cfa721195317",
                    "versions.yml:md5,16bf76d2f56399280213cfa721195317",
                    "versions.yml:md5,be2acf8efb04dc33562c9d00df7c3a50"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-27T17:51:10.73612885"
    },
    "Should run stringtie workflow with stringtie = true": {
        "content": [
            {
                "0": [
                    "stringtie.merged.gtf:md5,6da479298d73d5b3216d4e1576a2bdf4"
                ],
                "1": [
                    "versions.yml:md5,16bf76d2f56399280213cfa721195317",
                    "versions.yml:md5,16bf76d2f56399280213cfa721195317",
                    "versions.yml:md5,be2acf8efb04dc33562c9d00df7c3a50"
                ],
                "stringtie_gtf": [
                    "stringtie.merged.gtf:md5,6da479298d73d5b3216d4e1576a2bdf4"
                ],
                "versions": [
                    "versions.yml:md5,16bf76d2f56399280213cfa721195317",
                    "versions.yml:md5,16bf76d2f56399280213cfa721195317",
                    "versions.yml:md5,be2acf8efb04dc33562c9d00df7c3a50"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-27T17:50:57.437102535"
    }
}